tests/system/source/main.d

```diff
@@ -81,37 +111,179 @@ void main ()
+        Hash[] utxo_hashs;
+        Pair[] signature_noises;
+        Pair[] node_key_pairs;
+        Enrollment[] enrolls;
+
+        foreach (idx; 0 .. NodeCnt)
+        {
+            utxo_hashs ~= UTXOSet.getHash(hashFull(freeze_txs[idx]),0);
+            signature_noises ~= Pair.random;
+
+            Pair node_key_pair;
+            node_key_pair.v = secretKeyToCurveScalar(keypairs[idx].secret);
+            node_key_pair.V = node_key_pair.v.toPoint();
+            node_key_pairs ~= node_key_pair;
+
+            Enrollment enroll;
+            enroll.utxo_key = utxo_hashs[idx];
+            enroll.random_seed = hashFull(Scalar.random());
+            enroll.cycle_length = 1008;
+            enroll.enroll_sig = sign(node_key_pair.v, node_key_pair.V,
+                signature_noises[idx].V, signature_noises[idx].v, enroll);
+            enrolls ~= enroll;
+
+            clients[0].enrollValidator(enrolls[idx]);
+        }
+
+        Thread.sleep(500.msecs);
+
+        // Checking Enrolled
         size_t times; // Number of times we slept for 50 msecs
         foreach (idx, ref client; clients)
         {
-        Start:
-            Thread.sleep(50.msecs);
-            const height = client.getBlockHeight();
-            // Retry if we're too early
-            if (height < 1 && times++ < 100) goto Start;
-            const blocks = client.getBlocksFrom(0, 42);
-            writefln("[%s] getBlockHeight: %s", idx, height);
-            writefln("[%s] getBlocksFrom: %s", idx, blocks.map!prettify);
-            writeln("----------------------------------------");
-            assert(height == 1);
-            assert(blocks.length == 2);
-            if (idx != 0)
-                assert(blockHash == hashFull(blocks[1].header));
-            else
-                blockHash = hashFull(blocks[1].header);
-            times = 0;
+            foreach (utxo_hash; utxo_hashs)
+            {
+                assert(client.hasEnrollment(utxo_hash));
+                writefln("[%s] hasEnrollment: %s", idx, utxo_hash);
+            }
         }
     }
+
+    /// Make Transactions for block height 4
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        const Block[] blocks_3 = clients[0].getBlocksFrom(0,4);
+        assert(blocks_3[3].header.height == 3);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx = {
+                type: TxType.Payment,
+                inputs: [Input(hashFull(blocks_3[3].txs[idx]), 0)],
+                outputs: [Output(Amount.MinFreezeAmount, kp.address)]
+            };
+
+            auto signature = kp.secret.sign(hashFull(tx)[]);
+            tx.inputs[0].signature = signature;
+            clients[2].putTransaction(tx);
+        }
+
+        checkBlockHeight(4);
+    }
+}
+
+/// Check block generation
+private void checkBlockHeight (ulong height)
```

Comment:
It seems like this commit does two things: refactors existing code and adds things like `checkBlockHeight`, and then also adds a test for the Enrollment.

Ideally they would be in separate commits.

Reply:
It was separated by separate commits.